lib/workers/bin/provider_worker

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env ruby
+
+require File.expand_path("../../../config/environment", __dir__)
+
+ems_id = ENV["EMS_ID"]
+
+ems      = ExtManagementSystem.find(ems_id)
+all_emss = [ems] + ems.child_managers
+ems_opts = all_emss.map do |e|
+  e.attributes.merge(
+    "endpoints"       => e.endpoints,
+    "authentications" => e.authentications
+  )
+end
+
+provider_opts = {
+  :ems       => ems_opts,
+  :messaging => MiqQueue.messaging_client_options,
+  :settings  => Settings.to_h
+}
+
+# Create a temp file and immediately unlink it to create a
+# secure hidden file to be used for the child process' stdin.
```

Comment:
We might want a comment here about why we didn't use a pipe and/or why we chose a tempfile.

Reply:
:+1: done